From 6244221e0ae076be26528ec6278bd0d5fac6bcfd Mon Sep 17 00:00:00 2001
From: Liu Qian <qian.liu@ingenic.com>
Date: Thu, 25 Apr 2024 15:48:10 +0800
Subject: [PATCH] [debug] for mips uclibc compile without statx problem

---
 libmount/src/utils.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libmount/src/utils.c b/libmount/src/utils.c
index 50553f1..b8907fc 100644
--- a/libmount/src/utils.c
+++ b/libmount/src/utils.c
@@ -112,6 +112,7 @@ static int safe_stat(const char *target, struct stat *st, int nofollow)
 	memset(st, 0, sizeof(struct stat));
 
 #ifdef AT_STATX_DONT_SYNC
+#if !defined(__UCLIBC__) || !defined(_mips)
 	{
 		int rc;
 		struct statx stx = { 0 };
@@ -137,6 +138,7 @@ static int safe_stat(const char *target, struct stat *st, int nofollow)
 			return rc;
 	}
 #endif
+#endif
 
 #ifdef AT_NO_AUTOMOUNT
 	return fstatat(AT_FDCWD, target, st,
-- 
2.7.4

