From 896a9fbad5fbec04bbed15978d95970488c74905 Mon Sep 17 00:00:00 2001
From: Yolanda Lee <jianfang.li@ingenic.com>
Date: Tue, 3 Dec 2024 11:27:02 +0800
Subject: [PATCH] [Fix] the issue of unsupported wide character printing.

---
 libbb/printable_string.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libbb/printable_string.c b/libbb/printable_string.c
index a814fd0..8e48174 100644
--- a/libbb/printable_string.c
+++ b/libbb/printable_string.c
@@ -28,7 +28,9 @@ const char* FAST_FUNC printable_string2(uni_stat_t *stats, const char *str)
 		}
 		if (c < ' ')
 			break;
+#if 0
 		if (c >= 0x7f)
 			break;
+#endif
 		s++;
 	}
-- 
2.34.1

