diff --git a/spl/Makefile b/spl/Makefile
index bd24f96..682a9ac 100644
--- a/spl/Makefile
+++ b/spl/Makefile
@@ -232,6 +232,9 @@ $(obj)u-boot-spl.bin:	$(obj)u-boot-spl
 endif
 endif
 endif
+ifdef CONFIG_NAND_BUILTIN_PARAMS
+	$(OBJTREE)/tools/ingenic-tools/sfc_nand_builtin_params $@ $@ $(CONFIG_SPIFLASH_PART_OFFSET)
+endif
 
 GEN_UBOOT = \
 	cd $(obj) && $(LD) $(LDFLAGS) $(LDFLAGS_$(@F)) $(__START) \
