diff -Naur a/compiler/rustc_codegen_ssa/src/back/metadata.rs b/compiler/rustc_codegen_ssa/src/back/metadata.rs
--- a/compiler/rustc_codegen_ssa/src/back/metadata.rs	2024-03-17 19:56:40.334168502 +0800
+++ b/compiler/rustc_codegen_ssa/src/back/metadata.rs	2024-03-17 19:43:45.795989343 +0800
@@ -212,6 +212,9 @@
             if sess.target.options.cpu.contains("r6") {
                 e_flags |= elf::EF_MIPS_NAN2008;
             }
+            if sess.target.options.cpu.contains("r2") {
+                e_flags |= elf::EF_MIPS_NAN2008;
+            }
             e_flags
         }
         Architecture::Mips64 => {
diff -Naur a/compiler/rustc_codegen_ssa/src/target_features.rs b/compiler/rustc_codegen_ssa/src/target_features.rs
--- a/compiler/rustc_codegen_ssa/src/target_features.rs	2024-03-17 19:56:40.334168502 +0800
+++ b/compiler/rustc_codegen_ssa/src/target_features.rs	2024-03-17 19:43:45.799928543 +0800
@@ -238,6 +238,7 @@
     ("fp64", Some(sym::mips_target_feature)),
     ("msa", Some(sym::mips_target_feature)),
     ("virt", Some(sym::mips_target_feature)),
+    ("nan2008", Some(sym::mips_target_feature)),
     // tidy-alphabetical-end
 ];
 
diff -Naur a/compiler/rustc_target/src/spec/mipsel_unknown_linux_gnu.rs b/compiler/rustc_target/src/spec/mipsel_unknown_linux_gnu.rs
--- a/compiler/rustc_target/src/spec/mipsel_unknown_linux_gnu.rs	2024-03-17 19:56:40.259323705 +0800
+++ b/compiler/rustc_target/src/spec/mipsel_unknown_linux_gnu.rs	2024-03-17 19:43:45.390251756 +0800
@@ -9,7 +9,7 @@
 
         options: TargetOptions {
             cpu: "mips32r2".into(),
-            features: "+mips32r2,+fpxx,+nooddspreg".into(),
+            features: "+mips32r2,+fp64,+nan2008,+nooddspreg".into(),
             max_atomic_width: Some(32),
             mcount: "_mcount".into(),
 
