diff -Nuar b/lib/libv4l1/v4l1compat.c a/lib/libv4l1/v4l1compat.c
--- b/lib/libv4l1/v4l1compat.c	2023-02-25 00:40:23.000000000 +0800
+++ a/lib/libv4l1/v4l1compat.c	2024-01-17 16:05:11.210793259 +0800
@@ -20,10 +20,6 @@
  */
 
 /* ensure we see *64 variants and they aren't transparently used */
-#undef _LARGEFILE_SOURCE
-#undef _FILE_OFFSET_BITS
-#define _LARGEFILE64_SOURCE 1
-
 #include <config.h>
 #include <stdlib.h>
 #include <stdarg.h>
@@ -65,6 +61,7 @@
 }
 
 #if defined(linux) && defined(__GLIBC__)
+#if !(__GLIBC_PREREQ(2, 38))
 LIBV4L_PUBLIC int open64(const char *file, int oflag, ...)
 {
 	int fd;
@@ -85,6 +82,7 @@
 	return fd;
 }
 #endif
+#endif
 
 LIBV4L_PUBLIC int close(int fd)
 {
@@ -124,12 +122,14 @@
 }
 
 #if defined(linux) && defined(__GLIBC__)
+#if !(__GLIBC_PREREQ(2, 38))
 LIBV4L_PUBLIC void *mmap64(void *start, size_t length, int prot, int flags, int fd,
 		__off64_t offset)
 {
 	return v4l1_mmap(start, length, prot, flags, fd, offset);
 }
 #endif
+#endif
 
 LIBV4L_PUBLIC int munmap(void *start, size_t length)
 {
diff -Nuar b/lib/libv4l2/v4l2convert.c a/lib/libv4l2/v4l2convert.c
--- b/lib/libv4l2/v4l2convert.c	2023-02-25 00:40:23.000000000 +0800
+++ a/lib/libv4l2/v4l2convert.c	2024-01-17 16:06:12.005373714 +0800
@@ -23,11 +23,6 @@
 /* prevent GCC 4.7 inlining error */
 #undef _FORTIFY_SOURCE
 
-/* ensure we see *64 variants and they aren't transparently used */
-#undef _LARGEFILE_SOURCE
-#undef _FILE_OFFSET_BITS
-#define _LARGEFILE64_SOURCE 1
-
 #ifdef ANDROID
 #include <android-config.h>
 #else
@@ -98,6 +93,7 @@
 }
 
 #if defined(linux) && defined(__GLIBC__)
+#if !(__GLIBC_PREREQ(2, 38))
 LIBV4L_PUBLIC int open64(const char *file, int oflag, ...)
 {
 	int fd;
@@ -121,6 +117,7 @@
 	return fd;
 }
 #endif
+#endif
 
 #ifndef ANDROID
 LIBV4L_PUBLIC int close(int fd)
@@ -161,12 +158,14 @@
 }
 
 #if defined(linux) && defined(__GLIBC__)
+#if !(__GLIBC_PREREQ(2, 38))
 LIBV4L_PUBLIC void *mmap64(void *start, size_t length, int prot, int flags, int fd,
 		off64_t offset)
 {
 	return v4l2_mmap(start, length, prot, flags, fd, offset);
 }
 #endif
+#endif
 
 LIBV4L_PUBLIC int munmap(void *start, size_t length)
 {
